arkane/sensitivity.py

```diff
@@ -276,39 +277,78 @@ def __init__(self, job, output_directory, perturbation):
-        transition_states = []
+        base_wells = []
+        base_wells.extend(self.job.network.reactants)
+        base_wells.extend(self.job.network.isomers)
+        base_wells.extend(self.job.network.products)
+        base_transition_states = []
         for rxn in self.job.network.path_reactions:
             # if rxn.transition_state is not None:
-            transition_states.append(rxn.transition_state)
-
-        for entry in wells + transition_states:
-            if entry in wells:
-                logging.info("\n\nPerturbing well '{0}' by {1}:".format(entry, self.perturbation))
-            else:
-                logging.info("\n\nPerturbing TS '{0}' by {1}:".format(entry.label, self.perturbation))
-            self.perturb(entry)
-            self.job.execute(output_file=None, plot=False, print_summary=False)  # run the perturbed job
-            self.unperturb(entry)
+            base_transition_states.append(rxn.transition_state)
+        base_perturbation = deepcopy(self.perturbation)
+        base_job = self.job
+        for j in range(len(base_wells + base_transition_states)):
+            self.perturbation = base_perturbation
+            base_entry = (base_wells + base_transition_states)[j]
+            failed = False
+            c = 0
+            wells = []
+            transition_states = []
+            while c < self.max_iters:
+                self.job = deepcopy(base_job)
+                wells = []
+                wells.extend(self.job.network.reactants)
+                wells.extend(self.job.network.isomers)
+                wells.extend(self.job.network.products)
+                transition_states = []
+                for rxn in self.job.network.path_reactions:
+                    # if rxn.transition_state is not None:
+                    transition_states.append(rxn.transition_state)
+                entry = (wells+transition_states)[j]
+                if entry in wells:
+                    logging.info("\n\nPerturbing well '{0}' by {1}:".format(entry, self.perturbation))
+                else:
+                    logging.info("\n\nPerturbing TS '{0}' by {1}:".format(entry.label, self.perturbation))
+                self.perturb(entry)
+                try:
+                    self.job.execute(output_file=None, plot=False, print_summary=False)  # run the perturbed job
+                    self.unperturb(entry)
+                    break
+                except (InvalidMicrocanonicalRateError, ModifiedStrongCollisionError) as e:
+                    self.unperturb(entry)
+                    c += 1
+                    self.perturbation = quantity.Quantity(self.perturbation.value/2.0, self.perturbation.units)
+                    logging.error("Decreasing perturbation to {}".format(self.perturbation))
+                    
+            if c == self.max_iters:
+                if entry in wells:
+                    logging.error("Perturbation of well '{0}' has failed".format(entry))
```

Comment:
That's nice, but the other object read by the f-string is different so it would have to have another one of those if statements and I think that's too much for one line.